src/Compilers/Shared/BuildServerConnection.cs

```diff
@@ -506,17 +507,28 @@ internal static string GetPipeNameForPathOpt(string compilerExeDirectory)
             return $"{userName}.{isAdmin}.{basePipeName}";
         }
 
-        internal static string GetBasePipeName(string compilerExeDirectory)
+        internal static string GetBasePipeName(string compilerExeDirectory, string sharedCompilationId = null)
         {
             // Normalize away trailing slashes.  File APIs include / exclude this with no 
             // discernable pattern.  Easiest to normalize it here vs. auditing every caller
             // of this method.
             compilerExeDirectory = compilerExeDirectory.TrimEnd(Path.DirectorySeparatorChar);
 
+            // Store refined compilerExeDirectory value as the basis for a raw pipe name to be hashed.
+            string rawPipeName = compilerExeDirectory;
```

Comment:
I'm confused at what's going on here. Is the pipe name now the `path + sharedcompilationid`? If this is something the user explicitly specifies, wouldn't it be better to just use that value verbatim?